randovania/games/samus_returns/gui/hint_details_tab.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+import collections
+import random
+from typing import TYPE_CHECKING
+
+from PySide6 import QtWidgets
+
+from randovania.exporter.hints.hint_exporter import HintExporter
+from randovania.games.samus_returns.exporter.hint_namer import MSRHintNamer
+from randovania.gui.game_details.game_details_tab import GameDetailsTab
+from randovania.layout import filtered_database
+from randovania.lib.container_lib import iterate_key_sorted
+
+if TYPE_CHECKING:
+    from randovania.game_description.game_patches import GamePatches
+    from randovania.games.game import RandovaniaGame
+    from randovania.interface_common.players_configuration import PlayersConfiguration
+    from randovania.layout.base.base_configuration import BaseConfiguration
+
+
+class MSRHintDetailsTab(GameDetailsTab):
+    def __init__(self, parent: QtWidgets.QWidget, game: RandovaniaGame):
+        super().__init__(parent, game)
+        self.tree_widget = QtWidgets.QTreeWidget(parent)
+
+    def widget(self) -> QtWidgets.QWidget:
+        return self.tree_widget
+
+    def tab_title(self) -> str:
+        return "Hints"
+
+    def update_content(
+        self, configuration: BaseConfiguration, all_patches: dict[int, GamePatches], players: PlayersConfiguration
+    ):
+        self.tree_widget.clear()
+        self.tree_widget.setColumnCount(3)
+        self.tree_widget.setHeaderLabels(["Hint", "Pickup", "In-Game Text"])
+
+        game = filtered_database.game_description_for_layout(configuration)
+        region_list = game.region_list
+        patches = all_patches[players.player_index]
+
+        per_region: dict[str, dict[str, tuple[str, str]]] = collections.defaultdict(dict)
+        namer = MSRHintNamer(all_patches, players)
+        exporter = HintExporter(namer, random.Random(0), ["A joke hint."])
+
+        for identifier, hint in patches.hints.items():
+            node = game.region_list.node_by_identifier(identifier)
+            source_region = region_list.nodes_to_region(node)
+            source_name = region_list.node_name(node)
+
+            hint_text = exporter.create_message_for_hint(hint, all_patches, players, False)
```

Comment:
If you don't want to change the patcher now, use `.strip()` at the end of this line.
It should remove the new line and fix the padding.